### ISSUE TITLE
[blocker] adb defunct kills docker daemon on host

### ISSUE THREAD

**Issue by Kurisutian** [butomo1989/docker-android#86]
Operating System: 
Debian 8

Docker Image:
butomo1989/docker-android-x86-7.1.1

Docker Version:  
17.12.1-ce

Docker Command to start docker-android:  
Running everything through Rancher
  
## Expected Behavior
When running the containers for some hours or days they should keep running stable.

## Actual Behavior
After a while (sometimes hours sometimes days) I noticed that quite a few of our containers for automated testing (>40 instanced) started to show a huge amount of defunct adb processes in each container. This usually leads to the point that the whole docker daemon on the host gets killed, the host becomes unusable for all docker containers, independently of running privileged or not, and the daemon on the host gets so messed up that it even cannot be restarted any more as it simply locks up and refuses to start, stop or operate in any way. So the only way out is to hard restart the host by turining the power off and on. 
Because of this the containers cannot be used in our environment since we have quite a few huge tests that might run for a while and while running the container starts acting this way tearing down the host.

**Comment by JoeSSS** [comment]
I can imagine that [this is the problem](https://github.com/butomo1989/docker-android/blob/master/docker/Emulator_x86#L154) The healtcheck seems to be restarted as soon as adb is having troubles (tried to run `adb emu kill` in the container and healcheck started again automatically). And healtcheck spawns more and more instances:

```
root      2918  0.0  0.0   4624   804 ?        Ss   17:50   0:00 /bin/sh -c adb wait-for-device shell 'while [[ -z $(getprop sys.boot_completed) ]]; do sleep 1; done'
root      2924  0.0  0.0  18996  1632 ?        S    17:50   0:00 adb wait-for-device shell while [[ -z $(getprop sys.boot_completed) ]]; do sleep 1; done
root      2936  0.0  0.0   4624   800 ?        Ss   17:50   0:00 /bin/sh -c adb wait-for-device shell 'while [[ -z $(getprop sys.boot_completed) ]]; do sleep 1; done'
root      2941  0.0  0.0  18996  1576 ?        S    17:50   0:00 adb wait-for-device shell while [[ -z $(getprop sys.boot_completed) ]]; do sleep 1; done
root      2944  0.0  0.0   4624   812 ?        Ss   17:51   0:00 /bin/sh -c adb wait-for-device shell 'while [[ -z $(getprop sys.boot_completed) ]]; do sleep 1; done'
root      2950  0.0  0.0  18996  1564 ?        S    17:51   0:00 adb wait-for-device shell while [[ -z $(getprop sys.boot_completed) ]]; do sleep 1; done
```

@butomo1989 do you know why healthcheck strikes back? Reading the line of code it should exit after being healthy and never return.

**Comment by butomo1989** [comment]
Fixed in release [1.2](https://github.com/butomo1989/docker-android/releases/tag/1.2)
